bbb/oderhs.m

```diff
@@ -8093,9 +8224,14 @@ call s2fill (nx+2, ny+2, 0., volmsor(0:nx+1,0:ny+1,ifld), 1, nx+2)
         mvolcurt = mvolcurt + mvolcur(ifld)
       enddo
       do igsp = 1, ngsp
-        call s2fill (nx+2, ny+2, 0., volpsor(0:nx+1,0:ny+1,igsp), 1, nx+2)
-        call s2fill (nx+2, ny+2, 0., volmsor(0:nx+1,0:ny+1,igsp), 1, nx+2)
+        #..zmlpossible bugs?
+        #call s2fill (nx+2, ny+2, 0., volpsor(0:nx+1,0:ny+1,igsp), 1, nx+2) #..zml defined above
+        #call s2fill (nx+2, ny+2, 0., volmsor(0:nx+1,0:ny+1,igsp), 1, nx+2)
+	call s2fill (nx+2, ny+2, 0., volpsorg(0:nx+1,0:ny+1,igsp), 1, nx+2)
         ivolcurgt = ivolcurgt + ivolcurg(igsp)
+        #..zml manufactured solution
+        if (ismanufactured(igsp) .ne. 1)
+     .    call s2fill (nx+2, ny+2, 0., pwrsorg(0,0,igsp), 1, nx+2)
```

Comment:
Bill traced the error to newer compilers asserting the shape of the passed arguments. You need to pass the whole pwrsorg array rather than only the first index, e.g. pwrsorg(0:nx+1,0:ny+1,igsp)